### ISSUE TITLE
Unable to create images with perf test toolkit on sandbox 17.2

### ISSUE THREAD

**Issue by tfenster** [microsoft/navcontainerhelper#1617]
PLEASE DO NOT INCLUDE ANY PASSWORDS OR TOKENS IN YOUR ISSUE!!!

**Describe the issue**
When creating a new image (french localization) with the perf test toolkit for a sandbox using the currently latest artifacts (17.2), I get an error and the image can't be created:
```
Could not install the extension 'Performance Toolkit' by 'Microsoft' (version '17.2.19367.20828') on tenant 'default' and company '' due to the following error: 'Cannot install on environment that 
is not a Sandbox or OnPrem.//Please contact your administrator.' and AL stack trace:
"BCPT Install"(CodeUnit 149020).OnInstallAppPerDatabase line 5 - Performance Toolkit by Microsoft
"Extension Triggers"(CodeUnit 2000000010).OnInstallAppPerDatabase line 2
```

**Scripts used to create container and cause the issue**
License URL removed
```
import-module bccontainerhelper 
New-BcImage -artifactUrl (Get-BCArtifactUrl -country fr -storageAccount bcartifacts -type sandbox -version 17) -imageName bcartifacts/cosmo-bc -skipIfImageAlreadyExists -includeTestToolkit -includeTestLibrariesOnly -includePerformanceToolkit -licenseFile ... bcartifacts/cosmo-bc:sandbox-17.2.19367.20828-fr-mt 
```

**Full output of scripts**
License URL removed
```
PS C:\> import-module bccontainerhelper 
BcContainerHelper version 1.0.16 
WARNING: The names of some imported commands from the module 'bccontainerhelper' include unapproved verbs that might make them less discoverable. To find the commands with unapproved verbs, run  
the Import-Module command again with the Verbose parameter. For a list of approved verbs, type Get-Verb.
PS C:\> New-BcImage -artifactUrl (Get-BCArtifactUrl -country fr -storageAccount bcartifacts -type sandbox -version 17) -imageName bcartifacts/cosmo-bc -skipIfImageAlreadyExists -includeTestToolkit 
-includeTestLibrariesOnly -includePerformanceToolkit -licenseFile ... bcartifacts/cosmo-bc:sandbox-17.2.19367.20828-fr-mt 
Fetching all docker images 
Building multitenant image bcartifacts/cosmo-bc:sandbox-17.2.19367.20828-fr-mt based on mcr.microsoft.com/businesscentral:10.0.17763.1637 with https://bcartifacts.azureedge.net/sandbox/17.2.19367.2
0828/fr
Pulling latest image mcr.microsoft.com/businesscentral:10.0.17763.1637 
10.0.17763.1637: Pulling from businesscentral 
Generic Tag: 1.0.1.3 
Container OS Version: 10.0.17763.1637 (ltsc2019) 
Host OS Version: 10.0.17763.1637 (ltsc2019) 
Using process isolation
Using license file ...
Downloading c:\bcartifacts.cache\l11n4jjw.crl\my\license.flf 
Files in c:\bcartifacts.cache\l11n4jjw.crl\my: 
- license.flf
Copying Platform Artifacts 
Copying Database 
Copying Licensefile 
Copying ConfigurationPackages 
Copying Extensions 
Copying Applications.FR 
c:\bcartifacts.cache\l11n4jjw.crl 
Sending build context to Docker daemon   1.86GB 

Step 1/6 : FROM mcr.microsoft.com/businesscentral:10.0.17763.1637 
 ---> 72b1298716bc
Step 2/6 : ENV DatabaseServer=localhost DatabaseInstance=SQLEXPRESS DatabaseName=CRONUS IsBcSandbox=Y artifactUrl=https://bcartifacts.azureedge.net/sandbox/17.2.19367.20828/fr
 ---> Using cache
 ---> 50b3c085c0bf
Step 3/6 : COPY my /run/
 ---> Using cache
 ---> c35c21974f9e
Step 4/6 : COPY NAVDVD /NAVDVD/
 ---> 973a5418c18f 
Step 5/6 : RUN \Run\start.ps1 -installOnly -multitenant -includeTestToolkit -includeTestLibrariesOnly -includePerformanceToolkit
 ---> Running in 9049d701b277 
Using installer from C:\Run\150-new 
Installing Business Central 
Installing from DVD 
Starting Local SQL Server 
Starting Internet Information Server 
Copying Service Tier Files 
Copying PowerShell Scripts 
Copying dependencies 
Copying ReportBuilder 
Importing PowerShell Modules 
Determining Database Collation 
Changing Database Server Collation to French_100_CS_AS 
Restoring CRONUS Demo Database 
Setting CompatibilityLevel for tenant on localhost\SQLEXPRESS 
Exporting Application to CRONUS 
Removing Application from tenant 
Modifying Business Central Service Tier Config File for Docker 
Creating Business Central Service Tier 
Installing SIP crypto provider: 'C:\Windows\System32\NavSip.dll' 
Copying Web Client Files 
Copying Client Files
Copying ModernDev Files
Copying additional files
Copying ConfigurationPackages
Copying Test Assemblies
Copying Extensions
Copying Applications
Copying Applications.FR
Starting Business Central Service Tier 
Importing CRONUS license file 
Copying Database on localhost\SQLEXPRESS from tenant to default 
Taking database tenant offline 
Copying database files 
Attaching files as new Database default 
Putting database tenant back online 
Mounting tenant database 
Mounting Database for default on server localhost\SQLEXPRESS with AllowAppDatabaseWrite = False 
Sync'ing Tenant 
Tenant is Operational 
Publishing C:\Applications.FR\Microsoft_Any_17.2.19367.20828.app 
Synchronizing Any 
Installing Any 
Publishing C:\Applications.FR\Microsoft_Library Assert_17.2.19367.20828.app 
Synchronizing Library Assert 
Installing Library Assert 
Publishing C:\Applications.FR\Microsoft_Library Variable Storage_17.2.19367.20828.app 
Synchronizing Library Variable Storage 
Installing Library Variable Storage 
Test Runner is already published 
Synchronizing Test Runner
Installing Test Runner 
Publishing C:\Applications.FR\Microsoft_System Application Test Library_17.2.19367.20828.app 
Synchronizing System Application Test Library 
Installing System Application Test Library 
Publishing C:\Applications.FR\Microsoft_Tests-TestLibraries_17.2.19367.20828.app 
Synchronizing Tests-TestLibraries 
Installing Tests-TestLibraries 
Performance Toolkit is already published 
Synchronizing Performance Toolkit
Installing Performance Toolkit 
Could not install the extension 'Performance Toolkit' by 'Microsoft' (version '17.2.19367.20828') on tenant 'default' and company '' due to the following error: 'Cannot install on environment that 
is not a Sandbox or OnPrem.//Please contact your administrator.' and AL stack trace:
"BCPT Install"(CodeUnit 149020).OnInstallAppPerDatabase line 5 - Performance Toolkit by Microsoft
"Extension Triggers"(CodeUnit 2000000010).OnInstallAppPerDatabase line 2

Installation failed 
At C:\Run\start.ps1:368 char:9
+         throw "Installation failed"
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Installation failed:String) [
   ], RuntimeException
    + FullyQualifiedErrorId : Installation failed

The command 'powershell -Command $ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue'; \Run\start.ps1 -installOnly -multitenant -includeTestToolkit -includeTestLibrariesOnly -i
ncludePerformanceToolkit' returned a non-zero code: 1

Building image took 556 seconds 
...
```

**Additional context**
- does it happen all the time?
Yes, now it is completely reproducable
- did it use to work?
Yes

**Comment by freddydk** [comment]
I think this is the same bug - I have forwarded it to the team

**Comment by tfenster** [comment]
Only workaround for now is to remove the `-includePerformanceToolkit`?

**Comment by freddydk** [comment]
Yes - or grab the source from the perf toolkit and remove the check.

**Comment by tfenster** [comment]
The 17.3 artifacts have arrived, but are unfortunately still broken 😒

```
New-BcImage -artifactUrl (Get-BCArtifactUrl -country fr -storageAccount bcartifacts -type sandbox -version 17) -imageName bcartifacts/cosmo-bc -skipIfImageAlreadyExists -includeTestToolkit 
-includeTestLibrariesOnly -licenseFile ... -includePerformanceToolkit
Downloading application artifact /sandbox/17.3.20469.20825/fr 
Downloading C:\Users\ContainerAdministrator\AppData\Local\Temp\db3a3258-cd6c-4be1-b922-7a97d8a20e6c.zip
Unpacking application artifact to tmp folder using Expand-Archive 
Downloading platform artifact /sandbox/17.3.20469.20825/platform 
Downloading C:\Users\ContainerAdministrator\AppData\Local\Temp\63952236-ca68-499b-976d-c27fab845847.zip
Unpacking platform artifact to tmp folder using Expand-Archive 
Downloading Prerequisite Components 
Downloading c:\bcartifacts.cache\sandbox\17.3.20469.20825\platform\Prerequisite Components\Open XML SDK 2.5 for Microsoft Office\OpenXMLSDKv25.msi 
Downloading c:\bcartifacts.cache\sandbox\17.3.20469.20825\platform\Prerequisite Components\IIS URL Rewrite Module\rewrite_2.0_rtw_x64.msi 
Downloading c:\bcartifacts.cache\sandbox\17.3.20469.20825\platform\Prerequisite Components\DotNetCore\DotNetCore.1.0.4_1.1.1-WindowsHosting.exe 
bcartifacts/cosmo-bc:sandbox-17.3.20469.20825-fr-mt 
Fetching all docker images
Building multitenant image bcartifacts/cosmo-bc:sandbox-17.3.20469.20825-fr-mt based on mcr.microsoft.com/businesscentral:10.0.17763.1637 with https://bcartifacts.azureedge.net/sandbox/17.3.20469.2
0825/fr
Pulling latest image mcr.microsoft.com/businesscentral:10.0.17763.1637
10.0.17763.1637: Pulling from businesscentral 
Digest: sha256:3cec6e5344faf4bb70cabf8a57dab741afffab4b7f7f5f0ad42c3fe27fdfb926
Status: Image is up to date for mcr.microsoft.com/businesscentral:10.0.17763.1637
mcr.microsoft.com/businesscentral:10.0.17763.1637
Generic Tag: 1.0.1.3 
Container OS Version: 10.0.17763.1637 (ltsc2019) 
Host OS Version: 10.0.17763.1637 (ltsc2019)
Using process isolation
Using license file ...
Downloading c:\bcartifacts.cache\tbstaim0.lo1\my\license.flf
Files in c:\bcartifacts.cache\tbstaim0.lo1\my: 
- license.flf
Copying Platform Artifacts
Copying Database 
Copying Licensefile 
Copying ConfigurationPackages 
Copying Extensions 
Copying Applications.FR 
c:\bcartifacts.cache\tbstaim0.lo1 
Sending build context to Docker daemon  1.869GB 
 
Step 1/6 : FROM mcr.microsoft.com/businesscentral:10.0.17763.1637 
 ---> 72b1298716bc
Step 2/6 : ENV DatabaseServer=localhost DatabaseInstance=SQLEXPRESS DatabaseName=CRONUS IsBcSandbox=Y artifactUrl=https://bcartifacts.azureedge.net/sandbox/17.3.20469.20825/fr
 ---> Running in 5f43120fd21b 
Removing intermediate container 5f43120fd21b 
 ---> 11660d76c5ba
Step 3/6 : COPY my /run/
 ---> 83780feb99e1 
Step 4/6 : COPY NAVDVD /NAVDVD/
 ---> 57801b9734bd 
Step 5/6 : RUN \Run\start.ps1 -installOnly -multitenant -includeTestToolkit -includeTestLibrariesOnly -includePerformanceToolkit
 ---> Running in 7f6f7b775342 
Using installer from C:\Run\150-new 
Installing Business Central 
Installing from DVD 
Starting Local SQL Server
Starting Internet Information Server 
Copying Service Tier Files 
Copying PowerShell Scripts 
Copying dependencies 
Copying ReportBuilder 
Importing PowerShell Modules 
Determining Database Collation 
Changing Database Server Collation to French_100_CS_AS 
Restoring CRONUS Demo Database 
Setting CompatibilityLevel for tenant on localhost\SQLEXPRESS 
Exporting Application to CRONUS 
Removing Application from tenant 
Modifying Business Central Service Tier Config File for Docker 
Creating Business Central Service Tier 
Installing SIP crypto provider: 'C:\Windows\System32\NavSip.dll' 
Copying Web Client Files 
Copying Client Files
Copying ModernDev Files
Copying additional files
Copying ConfigurationPackages
Copying Test Assemblies
Copying Extensions
Copying Applications
Copying Applications.FR
Starting Business Central Service Tier 
Importing CRONUS license file 
Copying Database on localhost\SQLEXPRESS from tenant to default 
Taking database tenant offline 
Copying database files 
Attaching files as new Database default 
Putting database tenant back online 
Mounting tenant database 
Mounting Database for default on server localhost\SQLEXPRESS with AllowAppDatabaseWrite = False 
Sync'ing Tenant 
Tenant is Operational 
Publishing C:\Applications.FR\Microsoft_Any_17.3.20469.20825.app 
Synchronizing Any 
Installing Any 
Publishing C:\Applications.FR\Microsoft_Library Assert_17.3.20469.20825.app 
Synchronizing Library Assert 
Installing Library Assert 
Publishing C:\Applications.FR\Microsoft_Library Variable Storage_17.3.20469.20825.app 
Synchronizing Library Variable Storage 
Installing Library Variable Storage 
Test Runner is already published 
Synchronizing Test Runner
Installing Test Runner 
Publishing C:\Applications.FR\Microsoft_System Application Test Library_17.3.20469.20825.app 
Synchronizing System Application Test Library 
Installing System Application Test Library 
Publishing C:\Applications.FR\Microsoft_Tests-TestLibraries_17.3.20469.20825.app 
Synchronizing Tests-TestLibraries 
Installing Tests-TestLibraries 
Performance Toolkit is already published 
Synchronizing Performance Toolkit
Installing Performance Toolkit 
Could not install the extension 'Performance Toolkit' by 'Microsoft' (version '17.3.20469.20825') on tenant 'default' and company '' due to the following error: 'Cannot install on environment that 
is not a Sandbox or OnPrem.//Please contact your administrator.' and AL stack trace:
"BCPT Install"(CodeUnit 149020).OnInstallAppPerDatabase line 5 - Performance Toolkit by Microsoft
"Extension Triggers"(CodeUnit 2000000010).OnInstallAppPerDatabase line 2

Installation failed 
At C:\Run\start.ps1:368 char:9
+         throw "Installation failed"
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Installation failed:String) [
   ], RuntimeException
    + FullyQualifiedErrorId : Installation failed

The command 'powershell -Command $ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue'; \Run\start.ps1 -installOnly -multitenant -includeTestToolkit -includeTestLibrariesOnly -i
ncludePerformanceToolkit' returned a non-zero code: 1

Building image took 572 seconds 
```

**Comment by tfenster** [comment]
@freddydk is there any kind of ETA for this? We have a manual workaround but it starts to get annoying to do it all the time, so it would be nice to know whether we can expect a fix soon or not

**Comment by tfenster** [comment]
@freddydk Still no feedback? Would it help to create an official support ticket?

**Comment by freddydk** [comment]
I can see that the bug was fixed on January 25th.
I think it has been merged into the latest 17.2 and 17.3
Sorry for the lack of feedback, but there is no link between GitHub and our internal systems.

**Comment by tfenster** [comment]
On January 29th we still had the issue. I'll try again right now

**Comment by tfenster** [comment]
I am still getting this:

```
PS C:\> Import-Module -DisableNameChecking bccontainerhelper; New-BcImage -artifactUrl (Get-BCArtifactUrl -country fr -storageAccount bcartifacts -type sandbox -version 17) -imageName bcartifacts/cosmo-bc -skipIfImageAlreadyExists -includeTestToolkit -includeTestLibrariesOnly -includePerformanceToolkit -licenseFile <removed>
BcContainerHelper version 2.0.2
Downloading application artifact /sandbox/17.3.20469.21587/fr
Downloading C:\Users\ContainerAdministrator\AppData\Local\Temp\ea245355-0a0a-4a7c-9f2f-3fb3ade7b441.zip
Unpacking application artifact to tmp folder using Expand-Archive
Downloading platform artifact /sandbox/17.3.20469.21587/platform
Downloading C:\Users\ContainerAdministrator\AppData\Local\Temp\dcbb8196-23cd-44aa-bc80-392225de1ef7.zip
Unpacking platform artifact to tmp folder using Expand-Archive
Downloading Prerequisite Components
Downloading c:\bcartifacts.cache\sandbox\17.3.20469.21587\platform\Prerequisite Components\Open XML SDK 2.5 for Microsoft Office\OpenXMLSDKv25.msi
Downloading c:\bcartifacts.cache\sandbox\17.3.20469.21587\platform\Prerequisite Components\IIS URL Rewrite Module\rewrite_2.0_rtw_x64.msi
Downloading c:\bcartifacts.cache\sandbox\17.3.20469.21587\platform\Prerequisite Components\DotNetCore\DotNetCore.1.0.4_1.1.1-WindowsHosting.exe
bcartifacts/cosmo-bc:sandbox-17.3.20469.21587-fr-mt
Fetching all docker images
Building multitenant image bcartifacts/cosmo-bc:sandbox-17.3.20469.21587-fr-mt based on mcr.microsoft.com/businesscentral:10.0.19041.685 with https://bcartifacts.azureedge.net/s
andbox/17.3.20469.21587/fr
Pulling latest image mcr.microsoft.com/businesscentral:10.0.19041.685
10.0.19041.685: Pulling from businesscentral
Digest: sha256:2ab5cdca46deffa39ab0f2bb7d6e0fc594a5a68f176c3c24ce3902c352d170c0 
Status: Image is up to date for mcr.microsoft.com/businesscentral:10.0.19041.685
mcr.microsoft.com/businesscentral:10.0.19041.685
Generic Tag: 1.0.1.3
Container OS Version: 10.0.19041.685 (2004)
Host OS Version: 10.0.19041.685 (2004)
Using process isolation
Using license file <removed>
Downloading c:\bcartifacts.cache\qwj1wruv.lbw\my\license.flf
Files in c:\bcartifacts.cache\qwj1wruv.lbw\my:
- license.flf
Copying Platform Artifacts
Copying Database
Copying Licensefile
Copying ConfigurationPackages
Copying Extensions
Copying Applications.FR
c:\bcartifacts.cache\qwj1wruv.lbw
Sending build context to Docker daemon  1.869GB

Step 1/6 : FROM mcr.microsoft.com/businesscentral:10.0.19041.685
 ---> 896f16607421
Step 2/6 : ENV DatabaseServer=localhost DatabaseInstance=SQLEXPRESS DatabaseName=CRONUS IsBcSandbox=Y artifactUrl=https://bcartifacts.azureedge.net/sandbox/17.3.20469.21587/fr f
ilesOnly=False
 ---> Running in 89b0bae9a7f6
Removing intermediate container 89b0bae9a7f6
 ---> a0c134dd45da
Step 3/6 : COPY my /run/
 ---> 3e4e7168492a
Step 4/6 : COPY NAVDVD /NAVDVD/
 ---> c1c47669aafe
Step 5/6 : RUN \Run\start.ps1 -installOnly -multitenant -includeTestToolkit -includeTestLibrariesOnly -includePerformanceToolkit
 ---> Running in f2ebe5a3f3c9
Using installer from C:\Run\150-new
Installing Business Central
Installing from DVD
Starting Local SQL Server
Starting Internet Information Server
Copying Service Tier Files
Copying PowerShell Scripts
Copying dependencies
Copying ReportBuilder
Importing PowerShell Modules
Determining Database Collation
Changing Database Server Collation to French_100_CS_AS
Restoring CRONUS Demo Database
Setting CompatibilityLevel for tenant on localhost\SQLEXPRESS
Exporting Application to CRONUS
Removing Application from tenant
Modifying Business Central Service Tier Config File for Docker
Creating Business Central Service Tier
Installing SIP crypto provider: 'C:\Windows\System32\NavSip.dll'
Copying Web Client Files
Copying Client Files
Copying ModernDev Files
Copying additional files
Copying ConfigurationPackages
Copying Test Assemblies
Copying Extensions
Copying Applications
Copying Applications.FR
Starting Business Central Service Tier
Importing CRONUS license file
Copying Database on localhost\SQLEXPRESS from tenant to default
Taking database tenant offline
Copying database files
Attaching files as new Database default
Putting database tenant back online
Mounting tenant database
Mounting Database for default on server localhost\SQLEXPRESS with AllowAppDatabaseWrite = False
Sync'ing Tenant
Tenant is Operational
Publishing C:\Applications.FR\Microsoft_Any_17.3.20469.21587.app
Synchronizing Any
Installing Any
Publishing C:\Applications.FR\Microsoft_Library Assert_17.3.20469.21587.app
Synchronizing Library Assert
Installing Library Assert
Publishing C:\Applications.FR\Microsoft_Library Variable Storage_17.3.20469.21587.app
Synchronizing Library Variable Storage
Installing Library Variable Storage
Test Runner is already published
Synchronizing Test Runner
Installing Test Runner
Publishing C:\Applications.FR\Microsoft_System Application Test Library_17.3.20469.21587.app
Synchronizing System Application Test Library
Installing System Application Test Library
Publishing C:\Applications.FR\Microsoft_Tests-TestLibraries_17.3.20469.21587.app
Synchronizing Tests-TestLibraries
Installing Tests-TestLibraries
Performance Toolkit is already published
Synchronizing Performance Toolkit
Installing Performance Toolkit
Could not install the extension 'Performance Toolkit' by 'Microsoft' (version '17.3.20469.21587') on tenant 'default' and company '' due to the following error: 'Cannot install 
on environment that is not a Sandbox or OnPrem.//Please contact your administrator.' and AL stack trace:
"BCPT Install"(CodeUnit 149020).OnInstallAppPerDatabase line 5 - Performance Toolkit by Microsoft
"Extension Triggers"(CodeUnit 2000000010).OnInstallAppPerDatabase line 2

Installation failed
At C:\Run\start.ps1:368 char:9
+         throw "Installation failed"
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Installation failed:String) [
   ], RuntimeException
    + FullyQualifiedErrorId : Installation failed

The command 'powershell -Command $ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue'; \Run\start.ps1 -installOnly -multitenant -includeTestToolkit -include
TestLibrariesOnly -includePerformanceToolkit' returned a non-zero code: 1

Building image took 575 seconds
```

**Comment by freddydk** [comment]
Could you try in 17.2?

**Comment by tfenster** [comment]
The same result

```
PS C:\> Import-Module -DisableNameChecking bccontainerhelper; New-BcImage -artifactUrl (Get-BCArtifactUrl -country fr -storageAccount bcartifacts -type sandbox -version 17.2) -i
mageName bcartifacts/cosmo-bc -skipIfImageAlreadyExists -includeTestToolkit -includeTestLibrariesOnly -includePerformanceToolkit -licenseFile <removed>
Downloading application artifact /sandbox/17.2.19367.21585/fr
Downloading C:\Users\ContainerAdministrator\AppData\Local\Temp\82a9923f-b9eb-4743-aaa9-add21a9aa556.zip
Unpacking application artifact to tmp folder using Expand-Archive
Downloading platform artifact /sandbox/17.2.19367.21585/platform
Downloading C:\Users\ContainerAdministrator\AppData\Local\Temp\d8ca841f-c55a-4324-8828-3d999d549952.zip
Unpacking platform artifact to tmp folder using Expand-Archive
Downloading Prerequisite Components
Downloading c:\bcartifacts.cache\sandbox\17.2.19367.21585\platform\Prerequisite Components\Open XML SDK 2.5 for Microsoft Office\OpenXMLSDKv25.msi
Downloading c:\bcartifacts.cache\sandbox\17.2.19367.21585\platform\Prerequisite Components\IIS URL Rewrite Module\rewrite_2.0_rtw_x64.msi
Downloading c:\bcartifacts.cache\sandbox\17.2.19367.21585\platform\Prerequisite Components\DotNetCore\DotNetCore.1.0.4_1.1.1-WindowsHosting.exe
bcartifacts/cosmo-bc:sandbox-17.2.19367.21585-fr-mt
Fetching all docker images
Building multitenant image bcartifacts/cosmo-bc:sandbox-17.2.19367.21585-fr-mt based on mcr.microsoft.com/businesscentral:10.0.19041.685 with https://bcartifacts.azureedge.net/s
andbox/17.2.19367.21585/fr
Pulling latest image mcr.microsoft.com/businesscentral:10.0.19041.685
10.0.19041.685: Pulling from businesscentral
Digest: sha256:2ab5cdca46deffa39ab0f2bb7d6e0fc594a5a68f176c3c24ce3902c352d170c0
Status: Image is up to date for mcr.microsoft.com/businesscentral:10.0.19041.685
mcr.microsoft.com/businesscentral:10.0.19041.685
Generic Tag: 1.0.1.3
Container OS Version: 10.0.19041.685 (2004)
Host OS Version: 10.0.19041.685 (2004)
Using process isolation
Using license file <removed>
Downloading c:\bcartifacts.cache\qdt3ftvg.4lj\my\license.flf
Files in c:\bcartifacts.cache\qdt3ftvg.4lj\my:
- license.flf
Copying Platform Artifacts
Copying Database
Copying Licensefile
Copying ConfigurationPackages
Copying Extensions
Copying Applications.FR
c:\bcartifacts.cache\qdt3ftvg.4lj
Sending build context to Docker daemon  1.861GB

Step 1/6 : FROM mcr.microsoft.com/businesscentral:10.0.19041.685
 ---> 896f16607421
Step 2/6 : ENV DatabaseServer=localhost DatabaseInstance=SQLEXPRESS DatabaseName=CRONUS IsBcSandbox=Y artifactUrl=https://bcartifacts.azureedge.net/sandbox/17.2.19367.21585/fr f
ilesOnly=False
 ---> Running in ea378ef0a6e4
Removing intermediate container ea378ef0a6e4
 ---> 5fec4e4daa79
Step 3/6 : COPY my /run/
 ---> 66a980dd6362
Step 4/6 : COPY NAVDVD /NAVDVD/
 ---> ba099cc8747a
Step 5/6 : RUN \Run\start.ps1 -installOnly -multitenant -includeTestToolkit -includeTestLibrariesOnly -includePerformanceToolkit
 ---> Running in 04cb004fafa1
Using installer from C:\Run\150-new
Installing Business Central
Installing from DVD
Starting Local SQL Server
Starting Internet Information Server
Copying Service Tier Files
Copying PowerShell Scripts
Copying dependencies
Copying ReportBuilder
Importing PowerShell Modules
Determining Database Collation
Changing Database Server Collation to French_100_CS_AS
Restoring CRONUS Demo Database
Setting CompatibilityLevel for tenant on localhost\SQLEXPRESS
Exporting Application to CRONUS
Removing Application from tenant
Modifying Business Central Service Tier Config File for Docker
Creating Business Central Service Tier
Installing SIP crypto provider: 'C:\Windows\System32\NavSip.dll'
Copying Web Client Files
Copying Client Files
Copying ModernDev Files
Copying additional files
Copying ConfigurationPackages
Copying Test Assemblies
Copying Extensions
Copying Applications
Copying Applications.FR
Starting Business Central Service Tier
Importing CRONUS license file
Copying Database on localhost\SQLEXPRESS from tenant to default
Taking database tenant offline
Copying database files
Attaching files as new Database default
Putting database tenant back online
Mounting tenant database
Mounting Database for default on server localhost\SQLEXPRESS with AllowAppDatabaseWrite = False
Sync'ing Tenant
Tenant is Operational
Publishing C:\Applications.FR\Microsoft_Any_17.2.19367.21585.app
Synchronizing Any
Installing Any
Publishing C:\Applications.FR\Microsoft_Library Assert_17.2.19367.21585.app
Synchronizing Library Assert
Installing Library Assert
Publishing C:\Applications.FR\Microsoft_Library Variable Storage_17.2.19367.21585.app
Synchronizing Library Variable Storage
Installing Library Variable Storage
Test Runner is already published
Synchronizing Test Runner
Installing Test Runner
Publishing C:\Applications.FR\Microsoft_System Application Test Library_17.2.19367.21585.app
Synchronizing System Application Test Library
Installing System Application Test Library
Publishing C:\Applications.FR\Microsoft_Tests-TestLibraries_17.2.19367.21585.app
Synchronizing Tests-TestLibraries
Installing Tests-TestLibraries
Performance Toolkit is already published
Synchronizing Performance Toolkit
Installing Performance Toolkit
Could not install the extension 'Performance Toolkit' by 'Microsoft' (version '17.2.19367.21585') on tenant 'default' and company '' due to the following error: 'Cannot install 
on environment that is not a Sandbox or OnPrem.//Please contact your administrator.' and AL stack trace:
"BCPT Install"(CodeUnit 149020).OnInstallAppPerDatabase line 5 - Performance Toolkit by Microsoft
"Extension Triggers"(CodeUnit 2000000010).OnInstallAppPerDatabase line 2

Installation failed
At C:\Run\start.ps1:368 char:9
+         throw "Installation failed"
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Installation failed:String) [
   ], RuntimeException
    + FullyQualifiedErrorId : Installation failed

The command 'powershell -Command $ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue'; \Run\start.ps1 -installOnly -multitenant -includeTestToolkit -include
TestLibrariesOnly -includePerformanceToolkit' returned a non-zero code: 1

Building image took 552 seconds
```

**Comment by tfenster** [comment]
Fixed with the latest 17.3 artifacts! Thanks 👍
